web_client/views/popover/AnnotationPopover.js

```diff
@@ -0,0 +1,196 @@
+ */
+function rotation(r) {
+    return `${parseInt(r * 180 / Math.PI)}°`;
+}
+
+/**
+ * This view behaves like a bootstrap "popover" that follows the mouse pointer
+ * over the image canvas and dynamically updates according to the features
+ * under the pointer.
+ *
+ * @param {object} [settings]
+ * @param {number} [settings.debounce]
+ *   Debounce time in ms for rerendering due to mouse movements
+ */
+var AnnotationPopover = View.extend({
+    initialize(settings) {
+        if (settings.debounce) {
+            this._position = _.debounce(this._position, settings.debounce);
+        }
+
+        $('body').on('mousemove', '.h-image-view-body', (evt) => this._position(evt));
+        $('body').on('mouseout', '.h-image-view-body', () => this._hide());
+        $('body').on('mouseover', '.h-image-view-body', () => this._show());
+
+        this._hidden = !settings.visible;
+        this._users = {};
+        this.collection = new ElementCollection();
+        this.listenTo(this.collection, 'add', this._getUser);
+        this.listenTo(this.collection, 'all', this.render);
+    },
+
+    render() {
+        this.$el.html(
+            annotationPopover({
+                annotations: _.uniq(
+                    this.collection.pluck('annotation'),
+                    _.property('id')),
+                elements: this.collection.groupBy((element) => element.get('annotation').id),
+                formatDate: this._formatDate,
+                users: this._users,
+                elementProperties: this._elementProperties
+            })
+        );
+        this._show();
+        if (!this._visible()) {
+            this._hide();
+        }
+        this._height = this.$('.h-annotation-popover').height();
+    },
+
+    /**
+     * Set the popover visibility state.
+     *
+     * @param {boolean} [show]
+     *   if true: show the popover
+     *   if false: hide the popover
+     *   if undefined: toggle the popover state
+     */
+    toggle(show) {
+        if (show === undefined) {
+            show = this._hidden;
+        }
+        this._hidden = !show;
+        this.render();
+        return this;
+    },
+
+    /**
+     * Check the local cache for the given creator.  If it has not already been
+     * fetched, then send a rest request to get the user information and
+     * rerender the popover.
+     *
+     * As a consequence to avoid always rendering asyncronously, the user name
+     * will not be shown on the first render.  In practice, this isn't usually
+     * noticeable.
+     */
+    _getUser(model) {
+        var id = model.get('annotation').get('creatorId');
+        if (!_.has(this._users, id)) {
+            restRequest({
+                path: 'user/' + id
+            }).done((user) => {
+                this._users[id] = user;
+                this.render();
+            });
+        }
+    },
+
+    /**
+     * Format a Date object as a localized string.
+     */
+    _formatDate(s) {
```

Comment:
Is there a reason `_formatDate` is a view method whereas `point`, `length`, and `rotation` are module functions?